Code/Tools/FBuild/FBuildCore/Protocol/Protocol.h

```diff
@@ -32,7 +32,7 @@ namespace Protocol
 
     // Protocol Version
     enum : uint32_t { PROTOCOL_VERSION_MAJOR = 22 };    // Changes here make workers incompatible
-    enum : uint8_t  { PROTOCOL_VERSION_MINOR = 2 };     // Changes must be forwards and backwards compatible
+    enum : uint8_t  { PROTOCOL_VERSION_MINOR = 3 };     // Changes must be forwards and backwards compatible
```

Comment:
This changes the minor version of the protocol, but none of the logic in this PR uses that version to ensure backwards/forwards compatibility.

If a client was built with this code, and connected to a worker of a previous version it would crash trying to extract the json file from the payload in Client.cpp as it would not be present in the payload.

Similarly, a worker built with this version should not attach the new file for an older client when it doesn't expect that file to be attached.

Reply:
It seems that the server never gives its protocol version to the client so it will be difficult to ensure the backwards/forwards compatibility for the worker. Do you have any ideas on how to do this ?

Reply:
Hi, we still have not found on how to do this. Maybe you have an idea ?